In [68]:
import numpy as np
from scipy import stats

## Задача 5-1.
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,если выборочная средняя M = 80, а объем выборки n = 256.

### Решение:
Найдем доверительный интервал согласно формуле:$$P \left(\overline{X} + Z_{\dfrac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}} 
\leq M(X) \leq \overline{X} + Z_{\dfrac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}}\right)$$

где $\overline{X} = 80$, $\sigma = 16$, $n = 256$, а $Z$ найдем по таблице значений: $Z = 1,96$

In [69]:
std_ = 16
mean_ = 80
n = 256
print ('Доверительный интервал P =', [mean_ - 1.96*std_/np.sqrt(n), mean_ + 1.96*std_/np.sqrt(n)])

Доверительный интервал P = [78.04, 81.96]


#### Ответ 5-1: Доверительный интервал P = [78.04, 81.96]

## Задача 5-2.
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

### Решение:
[1] Найдем доверительный интервал согласно формуле:
$M \pm t_{\frac \alpha 2} \cdot \dfrac \sigma {\sqrt{n}}$, где:
- $n = 10$
- $1-\alpha = 95\%$
- $\alpha = 5\%$

In [70]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = 10
m = round((x.mean()), 2)
print(f'Математическое ожидание M(X) = {m}')
sigma = round((np.sqrt(((x - m)**2).sum()/(n-1))), 3)
print (f'Среднее квадратичное отклонение 𝜎 = {sigma}')

Математическое ожидание M(X) = 6.59
Среднее квадратичное отклонение 𝜎 = 0.451


[2] Табличное значение коэффициента Стьюдента:

In [71]:
t = 2.262
print (f'Коэффициент Стьюдента t = {t}')

Коэффициент Стьюдента t = 2.262


In [72]:
print ('Доверительный интервал P =', [round((m - (t*sigma/np.sqrt(n))), 5), round((m + (t*sigma/np.sqrt(n))), 5)])

Доверительный интервал P = [6.2674, 6.9126]


#### Ответ 5-2: Доверительный интервал, построенный с учётом доверительной вероятности 95%, P = [6.26754, 6.91246]. Истинное значение величины лежит пределах этого доверительного интервала.

## 3 и 4 задачи решать через тестирование гипотезы

## Задача 5-3.
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр $17~мм$. Используя односторонний критерий с $\alpha = 0,05$, проверить эту гипотезу, если в выборке из $n=100$ шариков средний диаметр оказался равным $17,5~мм$, а дисперсия известна и равна $4~кв.мм$.

### Решение:
[1] Сформулируем нулевую и альтернативную гипотезы:
- Гипотеза $H_0$: средний диаметр шарика для подшипников равен 17 мм ($\mu = \mu_0$).

- Гипотеза $H_1$: средний диаметр шарика для подшипников больше 17 мм ($\mu > \mu_0$).

[2] Зададим некоторую функцию Т от выборки:
Предполагается, что количество шариков большое, то распределение диаметра шариков является нормальным или близким к нему. Т.к. случайная величина диаметра имеет нормальное распределение, то среднее значение повыборке $T$ будет также иметь нормальное распределение с параметрами $a = 17$ и $\sigma = \sqrt{\dfrac{D(X)}{n}} = \sqrt {\dfrac{4}{100}} = 0.2 $

In [73]:
mu_0 = 17
mu = 17.5
n = 100
D = 4
sigma = np.sqrt(D/n)
print(f'Среднее квадратичное отклонение 𝜎 = {sigma}')

Среднее квадратичное отклонение 𝜎 = 0.2


[3] Фиксируем уровень значимости $\alpha$:
- допустимая для данной задачи вероятность ошибки первого рода (чаще всего 0.01, 0.05 или 0.1):

$\alpha = 0,05$

[4] Определяеем критическую область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$.
Выбранный уровень значимости $\alpha = 0,025$ означает, что для проверки гипотезы о среднем значении a = 17 требуется, чтобы с вероятностью 95% выборочное среднее попадало в правосторонний интервал, который (по правилу 2-х сигм) равен $(a - 2 \cdot \sigma,  a + 2 \cdot \sigma)$.

То есть критическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее = 17,5 попадёт в критическую область, то основная гипотеза отвергается.

In [74]:
print(f'Критическая область  Ω𝛼 = {[mu_0 - 2 * sigma, mu_0 + 2 * sigma]}')

Критическая область  Ω𝛼 = [16.6, 17.4]


[5] Проводем статистический тест:
для конкретной выборки $X$ считаем значение $T(X)$ , и если оно принадлежит $\Omega_\alpha$, то заключаем, что данные противоречат гипотезе $H_0$, и принимается гипотеза $H_1$.

In [75]:
stats.norm.ppf(1 - 0.025, 17, 0.2)
print(f'Статистический тест: {round((stats.norm.ppf(1 - 0.025, 17, 0.2)), 3)}')

Статистический тест: 17.392


$\color{red}{\textbf {UPD}}$ Коррекция решения:

[2] Табличное значение $Z$-критерия для $1-{\alpha}=1-0.05=0.95$ составляет $Z_T=1.645$

[3] Найдем значение $Z$-критерия для выборки по формуле: $Z_H=\dfrac{\mu-\mu_0}\sigma$:

In [76]:
z_obs = (mu - mu_0) / sigma
print(f'𝑍-критерий для выборки 𝑍_н = {z_obs}')

𝑍-критерий для выборки 𝑍_н = 2.5


[4] $Z_H=2,5 > Z_T=1,645$ следовательно при заданном уровне значимости $\alpha = 0,05$ верна альтернативная гипотеза $H_1$: средний диаметр шарика для подшипников больше 17 мм ($\mu > \mu_0$)

#### Ответ 5-3: При уровне значимости $\alpha = 0,05$ верна альтернативная гипотеза $H_1$: средний диаметр шарика для подшипников больше 17 мм.

## Задача 5-4.
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.
Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

### Решение:
[1] Найдем доверительный интервал, чтобы проверить гипотезу, что среднее значение веса пачки 200 г:

In [77]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mu = sample.mean()
n = 10
sigma = round((np.sqrt(((sample - mu)**2).sum()/(n-1))), 4)
print ('Среднее квадратичное отклонение 𝜎 = ', sigma)

Среднее квадратичное отклонение 𝜎 =  4.4535


[2] Табличное значение коэффициента Стьюдента:

In [78]:
t_int = 2.8214

[3] Проверим, что коэффицент Стьюдента = 2.8214, соответствует вероятности:

In [79]:
t = stats.t(n-1)
round((t.cdf(2.8214)), 7)

0.9899994

In [80]:
print ('Доверительный интервал P =', [round((mu - t_int*sigma/np.sqrt(n)), 3), round((mu + t_int*sigma/np.sqrt(n)), 3)])

Доверительный интервал P = [194.527, 202.473]


$\color{red}{\textbf {UPD}}$ Коррекция решения:

Известные величины:

- $\mu_0 = 200$
- $n = 10$
- $\alpha = 1\%$
- $1-\alpha = 99\%$


[1] Сформулируем нулевую и альтернативную гипотезы:
- Гипотеза $H_0$: средний вес пачки печенья равен 200 г ($\mu = \mu_0$).

- Гипотеза $H_1$: средний вес пачки печенья не равен 200 г ($\mu \ne \mu_0$).

[2] Так как выборка $n=10$ используем $t$-критерий Стьюдента. Для проверки гипотез сравним табличное и наблюдаемое значения $t$-критерия Стьюдента.

[3] Количество степеней свободы $v=n-1=10-1=9$. Табличное значение $t$-критерия Стьюдента для $\dfrac\alpha 2=\dfrac{0.01}2=0.005$ при $v=9$ составляет $t_T=t_{\frac \alpha 2}=\pm3.25$. Таким образом гипотеза $H_0$ верна при наблюдаемых значениях $t$-критерия Стьюдента находящихся в интервале [-3.25; 3.25].

[3] Найдем наблюдаемое значение $t$-критерия Стьюдента для выборки по формуле: $t_H=\dfrac{\mu-\mu_0}{\dfrac {\sigma_H} {\sqrt{n}}}$:

In [81]:
print(f'Математическое ожидание выборки 𝜇 = {mu}')
sigma_obs = np.sqrt(sample.var(ddof=1))
print ('Несмещенное среднее квадратичное отклонение 𝜎_н = ', round(sigma_obs, 4))
mu_0 = 200
t_H = ((mu - mu_0) * np.sqrt(n)) / sigma_obs
print ('Наблюдаемое значение 𝑡_н = ', round(t_H, 4))

Математическое ожидание выборки 𝜇 = 198.5
Несмещенное среднее квадратичное отклонение 𝜎_н =  4.4535
Наблюдаемое значение 𝑡_н =  -1.0651


[4] $t_H=-1.0651 \in [-3.25; 3.25]$ следовательно при заданном уровне значимости $\alpha = 0,01$ верна нулевая гипотеза $H_0$: средний вес пачки печенья равен 200 г ($\mu = \mu_0$)

#### Ответ 5-4: При уровне значимости  $\alpha = 0,01$ верна нулевая гипотеза $H_0$: средний вес пачки печенья равен 200 г. Утверждение продавца верно.